### PR TITLE
refactor: init hook `useRouterTargetWorkspace`

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

File: apps/web/src/pages/workspace/index.tsx [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
+    () =>
+      workspaceId &&
+      dataCenter.workspaces.length &&
+      dataCenter.workspaces.findIndex(
+        meta => meta.id.toString() === workspaceId
+      ) === -1,
+    [dataCenter.workspaces, workspaceId]
   );
-  const { workspaceLoaded } = useEnsureWorkspace();
+  if (notExist) {
```
Comment (pengx17) [inline apps/web/src/pages/workspace/index.tsx:25]: better put side effects in useEffect in case it will trigger multiple times because of rerendering

**Review by pengx17 — APPROVED** [review]
LGTM

File: apps/web/src/pages/workspace/[workspaceId]/index.tsx [hunk 2]
```diff
@@ -1,44 +1,48 @@
...
 const WorkspaceIndex = () => {
   const router = useRouter();
-  const currentWorkspace = useGlobalState(
-    useCallback(store => store.currentDataCenterWorkspace, [])
-  );
+  const { targetWorkspace, exist } = useRouterTargetWorkspace();
   const { createPage } = usePageHelper();
-  const { workspaceLoaded, activeWorkspaceId } = useEnsureWorkspace();
 
   useEffect(() => {
```
Comment (himself65) [inline apps/web/src/pages/workspace/[workspaceId]/index.tsx:13]: this place need to be rewritten in the future

**Comment by himself65** [comment]
some issue here

**Pull request merged**